### ISSUE TITLE
Tuple/struct property assignment fails (doesn't crash, but assignment has no effect)

### ISSUE THREAD

**Issue by gopakumar-gg** [beefytech/Beef#1477]
``` csharp
	class Demo
	{
		public (int x, int y) pos {get; set; } = .{ x = -1, y = -1};

		public int Num {get; set;} = 0;

		public void DoIt()
		{
			Num = 50; // Works fine.  Num is not a tuple or struct property.
			Console.WriteLine($"Marks = {Num}"); // shows 50 as expected.

			pos.x = 10; // NO EFFECT
			pos.y = 20; // NO EFFECT
			Console.WriteLine($"x = {pos.x}, y = {pos.y}"); // Shows WRONG value. x = -1, y = -1

			// If pos was declared just as member variable, above assignments would work fine.
			// public (int x, int y) pos = .{ x = -1, y = -1}; 
		}
	}
	class Program
	{
		public static void Main()
		{
			Demo t = new Demo();
			t.DoIt();
			delete t;
		}
	}
```

**Comment by gopakumar-gg** [comment]
Tested with Nightly build Feb 22 (e5f280)

**Comment by disarray2077** [comment]
I'm almost sure that the issue is caused by the tuple being returned by value, try to replace `public (int x, int y) pos {get; set; } = .{ x = -1, y = -1};` with `public ref (int x, int y) pos {get; set; } = .{ x = -1, y = -1};`

**Comment by gopakumar-gg** [comment]
That too didn't work.  Still x = -1, y = -1 is printed.

**Comment by bfiete** [comment]
`ref` should have worked - that part actually is a bug in an optimization where we elide the `get` invocation calls when a type accesses its own properties.

**Comment by bfiete** [comment]
But, yeah, in the original code- that will also fail to work in C# and C++, with the difference that C# explicitly shows an error for this.

**Comment by bfiete** [comment]
Fixed at 06f4eb9576ad4688b5a2d23824105dc39db8c16d

The non-ref version will now produce a warning on assignment, and the ref version now works properly.
